### ISSUE TITLE
Why notify the desktop to be inactivate?

### ISSUE THREAD

**Issue by DarwinSenior** [jaagr/polybar#411]
So, here is my question, I noticed the function `rebuild_desktop_states` in `src/modules/xworkspaces.cpp` sets active desktop to empty? Why that is case? Should not the workspace that is active always be active until it is not active?

```cpp
  /**
   * Update active state of current desktops
   */
  void xworkspaces_module::rebuild_desktop_states() {
    for (auto&& v : m_viewports) {
      for (auto&& d : v->desktops) {
        if (d->index == m_current_desktop && d->state != desktop_state::ACTIVE) {
          d->state = desktop_state::ACTIVE;
        } else if (d->state == desktop_state::ACTIVE) {
          d->state = desktop_state::EMPTY; // why deactivate it?
        } else if (d->label) {
          continue;
        }
```

**Comment by jaagr** [comment]
The state might be named badly but active is actually the state for focused workspaces.

**Comment by DarwinSenior** [comment]
Is it possible to have a current workspace? I have just switched to xmonad and I find it most useful.

Like this:

```cpp
  /**
   * Update active state of current desktops
   */
  void xworkspaces_module::rebuild_desktop_states() {
    for (auto&& v : m_viewports) {
      for (auto&& d : v->desktops) {
        if (d->index == m_current_desktop) {
          d->state = desktop_state::ACTIVE;
        } else {
          d->state = desktop_state:EMPTY;
        }
   }
```

**Comment by NBonaparte** [comment]
Closing, as this should be fixed by #587. Not sure why it didn't automatically close.
